.github/workflows/ci.yml

```diff
@@ -1,55 +1,23 @@
-name: CI
-on:
-  push:
-    branches-ignore:
-      - 'master'
+on: [push, pull_request]
 
 jobs:
-  format:
-    name: Format
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run format check'
-        run: cargo fmt -- --check
-  dependencies:
-    name: Dependencies
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run package check'
-        run: cargo check --all-targets --all-features
-  linter:
-    name: Linter
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run linter check'
-        run: cargo clippy --all-targets --all-features -- -D warnings
-  unittest:
-    name: Unittest
-    runs-on: ubuntu-latest
-    steps:
-      - name: 'Install Rust toolchain'
-        run: curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-      - name: 'Set PATH variable'
-        run: source $HOME/.cargo/env
-      - name: 'Fetch source code from repository'
-        uses: actions/checkout@v2
-      - name: 'Run unit tests'
-        run: cargo test --all-targets --all-features
+    ci:
+        strategy:
+            matrix:
+                rust: [stable, beta]
+        runs-on: ubuntu-latest
+        steps:
+            - name: Install Rust toolchain
+              uses: actions-rs/toolchain@v1
```

Comment:
I didn't use the `actions-rs/toolchain@v1`, because it is not from the Rust team. I wanted to get rid of dependencies. This is up for discussion of course.

Reply:
Imo it just makes it easier and less verbose.
And also it seems very well supported and updated.